cli/dcoscli/util.py

```diff
@@ -86,6 +86,46 @@ def confirm_text(prompt, confirmation_text):
     return False
 
 
+def choicelist(choices, descriptions=[], msg='Please choose: '):
+    """
+    Allow to pick an item from a choicelist. For example :
+
+        Please choose a login provider for your linked cluster:
+        1) dcos-uid-password
+        2) saml-sp-initiated
+        (1-2): [...]
+
+    :param choices: choices for the list
+    :type choices: list of str
+    :param descriptions: descriptions for the choices
+    :type descriptions: list of str
+    :returns: the chosen item
+    :rtype: str
+    """
+
+    if not descriptions:
+        descriptions = choices
+
+    if len(choices) != len(descriptions):
+        msg = 'Count mismatch between choices and descriptions.'
+        raise errors.DCOSException(msg)
+
+    emitter.publish(msg)
+    for i, desc in enumerate(descriptions):
+        emitter.publish('{}) {}'.format(i+1, desc))
+
+    # Write to stdout directly not to have a new line.
+    sys.stdout.write('({}-{}): '.format(1, len(choices)))
+    sys.stdout.flush()
+
+    try:
+        chosen_id = int(_read_response())
+    except ValueError:
+        return None
+
+    return choices[chosen_id-1] if len(choices) >= chosen_id else None
```

Comment:
Can you reverse the logic here so that it's `chosen_id <= len(choices)`. That is the standard ordering of inequalities for things like this and it makes it clearer at a glance what is going on.